pkgs/development/python-modules/pygam/default.nix

```
@@ -0,0 +1,29 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, future
+, numpy
+, pandas
+, progressbar2
+, scipy
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "pygam";
+  version = "0.8.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rli8w0fgfr6mxk5cx3n6i5zafi1j0abl2nslrrdxzmjm2p03bjw";
+  };
+
+  propagatedBuildInputs = [ future numpy pandas progressbar2 scipy ];
```

future package is only needed for python2
```suggestion
  propagatedBuildInputs = [ numpy pandas progressbar2 scipy ] ++ lib.optional isPy27 future;
```

---

Unfortunately it seems that the original library lists `future` as a requirement for all Python versions: https://github.com/dswah/pyGAM/blob/master/flit.ini#L16 .

After implementing the `isPy27` change, I receive the following error when running `nix-build . -A python3Packages.pygam`: 

```
Collecting future (from pygam==0.8.0)
  ERROR: Could not find a version that satisfies the requirement future (from pygam==0.8.0) (from versions: none)
ERROR: No matching distribution found for future (from pygam==0.8.0)
builder for '/nix/store/z6pcixliz8xlrbhwd76i074r0s36m59m-python3.7-pygam-0.8.0.drv' failed with exit code 1
error: build of '/nix/store/z6pcixliz8xlrbhwd76i074r0s36m59m-python3.7-pygam-0.8.0.drv' failed
```

Please advise.

---

oh, then add it back, and just leave a comment to the effect of `unfortunately future is listed in install_requires`

---

otherwise LGTM